### ISSUE TITLE
Mapping string.Empty to NULL

### ISSUE THREAD

**Issue by petervyvey** [agileobjects/AgileMapper#111]
Hi,

In a project it's needed to map all string.Empty values to NULL. I have tried some configurations, variations like the following:

            this.WhenMapping
                .From<string>()
                .To<string>()
                .After.MappingEnds
                .Call((src, dst) =>
                {
                    dst = string.IsNullOrEmpty(src) ? null : src;
                });

However none of them seem to work. Can you put in the right direction?

Thx,

Peter

**Comment by SteveWilkes** [comment]
Hi!

I don't think there's a way to do this out of the box, but I don't think it'd be too difficult to support:

```
Mapper.WhenMapping
    .From<string>().To<string>()
    .If(ctx => string.IsNullOrWhiteSpace(ctx.Source))
    .Map(default(string))
    .ToTarget();
```

I'll implement this, add it to the current pre-release and get back to you :)

Thanks for the feedback!

Steve

**Comment by SteveWilkes** [comment]
This is supported in v1.1 preview 3, which is now available on NuGet!

You could use:

```C#
Mapper.WhenMapping
    .From<string>().To<string>()
    .If(ctx => string.IsNullOrWhiteSpace(ctx.Source))
    .Map(default(string))
    .ToTarget();
```

...or:

```C#
Mapper.WhenMapping
    .From<string>().To<string>()
    .Map(ctx => string.IsNullOrWhiteSpace(ctx.Source) ? null : ctx.Source)
    .ToTarget();
```

Thanks again for the feedback!

Steve

**Comment by SteveWilkes** [comment]
Automatic operator use and simple-Type `ToTarget()` configuration are included in [v1.1](https://www.nuget.org/packages/AgileObjects.AgileMapper).

Thanks! :)
